build/builder.go

```diff
@@ -997,13 +1038,54 @@ func sortDocuments(todo []*zoekt.Document) {
 }
 
 func (b *Builder) buildShard(todo []*zoekt.Document, nextShardNum int) (*finishedShard, error) {
-	if !b.opts.DisableCTags && b.opts.CTagsPath != "" {
-		err := ctagsAddSymbols(todo, b.parser, b.opts.CTagsPath)
-		if b.opts.CTagsMustSucceed && err != nil {
-			return nil, err
+	universal := make([]*zoekt.Document, 0)
+	scip := make([]*zoekt.Document, 0)
+
+	for _, doc := range todo {
+		// This is also checked later in `shardBuilder.Add`, but we need it now select the right ctags parser
```

Comment:
this code living here feels a little strange. Maybe my previous suggestion around a smarter parser will fix this issue? If we still need this sort of logic, would be great to pull out into a function so we don't add so much knowledge about this aspect into buildShard directly

Reply:
Agreed, I'll see where the parser work leads me and if this still remains I'll split this into its own function.